airbyte-integrations/connectors/source-hubspot/source_hubspot/spec.json

```diff
@@ -15,16 +15,77 @@
-          "api_key": {
-            "description": "Hubspot API Key. See our <a href=\"https://docs.airbyte.io/integrations/sources/hubspot\">docs</a> if you need help finding this key.",
-            "type": "string",
-            "airbyte_secret": true
+        "oneOf": [
+          {
+            "type": "object",
+            "title": "api key",
+            "required": ["api_key", "credentials_title"],
+            "properties": {
+              "credentials_title": {
+                "type": "string",
+                "const": "API Key Credentials",
+                "enum": ["API Key Credentials"],
+                "default": "API Key Credentials",
+                "order": 0
+              },
+              "api_key": {
+                "description": "Hubspot API Key. See our <a href=\"https://docs.airbyte.io/integrations/sources/hubspot\">docs</a> if you need help finding this key.",
+                "type": "string",
+                "airbyte_secret": true
+              }
+            }
+          },
+          {
+            "type": "object",
+            "title": "oauth",
+            "required": [
+              "redirect_uri",
+              "client_id",
+              "client_secret",
+              "refresh_token",
+              "access_token",
+              "credentials_title"
+            ],
+            "properties": {
+              "credentials_title": {
+                "type": "string",
+                "const": "OAuth Credentials",
+                "enum": ["OAuth Credentials"],
+                "default": "OAuth Credentials",
+                "order": 0
+              },
+              "client_id": {
+                "description": "Hubspot client_id. See our <a href=\"https://docs.airbyte.io/integrations/sources/hubspot\">docs</a> if you need help finding this id.",
+                "type": "string",
+                "examples": ["123456789000"]
+              },
+              "client_secret": {
+                "description": "Hubspot client_secret. See our <a href=\"https://docs.airbyte.io/integrations/sources/hubspot\">docs</a> if you need help finding this secret.",
+                "type": "string",
+                "examples": ["secret"],
+                "airbyte_secret": true
+              },
+              "refresh_token": {
+                "description": "Hubspot refresh_token. See our <a href=\"https://docs.airbyte.io/integrations/sources/hubspot\">docs</a> if you need help generating the token.",
+                "type": "string",
+                "examples": ["refresh_token"],
+                "airbyte_secret": true
+              }
+            }
           }
-        }
+        ]
       }
     }
+  },
+  "authSpecification": {
+    "auth_type": "oauth2.0",
+    "oauth2Specification": {
+      "rootObject": ["credentials", 1],
+      "oauthFlowInitParameters": [
+        ["credentials", "client_id"],
+        ["credentials", "client_secret"],
+        ["credentials", "refresh_token"]
```

Comment:
we don't need the `credentials` here because it's already expressed by the root object
also refresh token is an `oauthFlowOutputParameters`